etc/kayobe/kolla/config/keystone/wsgi-keystone.conf

```diff
@@ -20,29 +20,34 @@ TraceEnable off
     CustomLog "{{ keystone_log_dir }}/keystone-apache-public-access.log" logformat
 {% endraw %}
     # OpenID Connect configuration for EGI Check-in AAI
-    OIDCResponseType "code"
+    OIDCResponseType "id_token"
     OIDCClaimPrefix "OIDC-"
     OIDCClaimDelimiter ;
     OIDCScope "openid profile email refeds_edu"
-    OIDCProviderMetadataURL https://aai-dev.egi.eu/oidc/.well-known/openid-configuration
-    OIDCClientID {{ secrets_egi_client_id }}
-    OIDCClientSecret {{ secrets_egi_client_secret }}
-    OIDCCryptoPassphrase {{ secrets_egi_crypto_passphrase }}
-    OIDCRedirectURI http://{% raw %}{{ kolla_external_fqdn }}:{{ keystone_public_port }}{% endraw %}/v3/auth/OS-FEDERATION/websso/oidc/redirect
+    OIDCProviderMetadataURL https://idp.stackhpc.com/auth/realms/AlaSKA/.well-known/openid-configuration
+    OIDCClientID alaska-alt1-keystone
+    OIDCClientSecret {{ secrets_alaska_keycloak_client_secret }}
+    OIDCCryptoPassphrase {{ secrets_alaska_keycloak_crypto_passphrase }}
+    OIDCRedirectURI http://{% raw %}{{ kolla_external_fqdn }}:{{ keystone_public_port }}{% endraw %}/v3/auth/OS-FEDERATION/websso/mapped/redirect
 
     # OAuth for CLI access
-    OIDCOAuthIntrospectionEndpoint  https://aai-dev.egi.eu/oidc/introspect
-    OIDCOAuthClientID {{ secrets_egi_client_id }}
-    OIDCOAuthClientSecret {{ secrets_egi_client_secret }}
+    OIDCOAuthIntrospectionEndpoint  https://idp.stackhpc.com/auth/realms/AlaSKA/protocol/openid-connect/token/introspect
+    OIDCOAuthClientID alaska-alt1-keystone
+    OIDCOAuthClientSecret {{ secrets_alaska_keycloak_client_secret }}
 
-    <Location ~ "/v3/auth/OS-FEDERATION/websso/oidc">
-            AuthType  openid-connect
-            Require   valid-user
-    </Location>
+    <LocationMatch /v3/auth/OS-FEDERATION/identity_providers/.*?/protocols/mapped/auth>
+      AuthType openid-connect
+      Require valid-user
+      LogLevel debug
```

Comment:
I'm assuming this doesn't generate insane amounts of output?

Reply:
It's not too crazy, and t's isolated to users authenticating via an IdP.  Once I've finished messing with Keycloak I'll drop that statement anyway.